### PR TITLE
Argon support

### PR DESCRIPTION
**Pull request by dflemstr**
I happen to have a bunch of argon (and xenon) boards, so I would like to try to make this crate work with it.  I might add wifi support too before calling this PR done.  The PR depends on #1 since otherwise I was unable to build things with bobbin

### CODE REVIEW

File: xenon/Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [package]
-authors = ["Tad FIsher <tadfisher@gmail.com>"]
+authors = ["Tad Fisher <tadfisher@gmail.com>"]
```
Comment (tadfisher) [inline xenon/Cargo.toml:2]: Hah, thanks.

**Review by tadfisher — APPROVED** [review]
This looks good as-is. I can merge this and you can work on WiFi support for the Argon, or I can wait; your call.

**Comment by dflemstr** [comment]
Hey, thanks for the review.  I made some minor tweaks to the pin/peripheral mapping as part of the development of the Wifi driver, and I might have to continue to do so.  So if you merge this PR, expect another one with breaking changes.

For reference, my very WIP Wifi driver is available here: https://github.com/dflemstr/esp-at-rs

I pushed it for the sake of this PR; the repo I linked is very incomplete and actually I'm working on lower level libraries down the stack first to support the driver before continuing on it.

**Comment by dflemstr** [comment]
OK, I didn't make enough progress in a timely manner... Feel free to merge this as is!

File: argon/examples/blinky.rs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#![no_main]
+#![no_std]
+
+extern crate panic_halt;
+
+use particle_xenon::{prelude::*, Board};
```
Comment (oddstr13) [inline argon/examples/blinky.rs]:
```suggestion
use particle_argon::{prelude::*, Board};
```

**Review by oddstr13 — APPROVED** [review]
Tiny copypasta error in the argon blinky, but otherwise compiles fine.

Relevant compile warning for this PR:
> warning: output filename collision.
> The example target `blinky` in package `particle-xenon v0.0.1 (.../particle-rs/xenon)` has the same output filename as the example target `blinky` in package `particle-argon v0.0.1 (.../particle-rs/argon)`.
> Colliding filename is: .../particle-rs/target/thumbv7em-none-eabihf/debug/examples/blinky
> The targets should have unique names.
> Consider changing their names to be unique or compiling them separately.
> This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.

This warning is easily fixed by renaming blinky.rs to blinky_argon.rs and blinky_xenon.rs

File: gen3-common/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[package]
+authors = ["David Flemström <david.flemstrom@gmail.com>", "Tad Fisher <tadfisher@gmail.com>"]
+categories = ["embedded", "hardware-support", "no-std"]
+description = "Board support crate common types for the Particle Generation 3 devices (Argon, Boron, Xenon)"
+keywords = ["arm", "cortex-m", "nrf52840", "hal"]
+license = "MIT OR Apache-2.0"
+name = "particle-gen3-common"
+repository = "https://github.com/tadfisher/particle-rs"
+documentation = "https://docs.rs/particle-argon/0.0.1/particle_argon/"
```
Comment (oddstr13) [inline gen3-common/Cargo.toml]:
```suggestion
documentation = "https://docs.rs/particle-gen3-common/0.0.1/particle_gen3_common/"
```

**Review by oddstr13 — APPROVED** [review]

**Comment by oddstr13** [comment]
@tadfisher I think this PR should be fine to merge now, do you have a few moments to do the merge bit?

**Comment by tadfisher** [comment]
Done, thanks!

**Pull request merged**